ehsm_kms_service/secret_manager_apis.js

```diff
@@ -0,0 +1,279 @@
+const crypto = require('crypto')
+const logger = require('./logger')
+const { v4: uuidv4 } = require('uuid')
+const {
+    SM_SECRET_VERSION_STAGE_CURRENT
+} = require('./constant')
+const {
+    napi_result,
+    _result,
+    base64_encode,
+    base64_decode
+} = require('./function')
+const {
+    cryptographic_apis,
+} = require('./apis')
+const ehsm_napi = require('./ehsm_napi')
+
+//check tags format
+function buildTagsArray(tags) {
+    try {
+        var tags_jsonArray = eval(tags)
+        if (!tags_jsonArray instanceof Array) {
+            return false
+        }
+        for (let i in tags_jsonArray) {
+            let tags_key = tags_jsonArray[i]['tagKey']
+            let tags_value = tags_jsonArray[i]['tagValue']
+            if (typeof tags_key != 'string' || tags_key.length == 0) {
+                return false
+            } else if (typeof tags_value != 'string' || tags_value.length == 0) {
+                return false
+            }
+        }
+        return tags_jsonArray
+    }
+    catch (e) {
+        return false
+    }
+}
+
+// get a string parameter from payload and base64 if you need
+function getParam_String(payload, key, needBase64) {
+    if (needBase64 == undefined) {
+        needBase64 = true;
+    }
+    let val = payload[key]
+    if (val != undefined) {
+        val = String(val)
+        if (needBase64 && val.length > 0) {
+            if (isJSONString(val)) {
+                val = toBase64(eval("(" + val + ")"))
```

Comment:
eval is dangerous, with high-security concerns. 
Please review the below and replace it with others, like Function().
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/eval#never_use_eval!

Reply:
will be change to Function